### ISSUE TITLE
Support case-insensitive search queries

### ISSUE THREAD

**Issue by Niush** [lorisleiva/laravel-search-string#22]
The %% Like searchable is not working.
`User::usingSearchString('Apple')->toSql();`
gives `select * from "users"`

But, this works
```php
return User::where(function ($query) {
    $query->where('name', 'like', '%Apple%')
          ->orWhere('address', 'like', '%Apple%');
})->toSql();
```
gives `select * from "users" where ("name"::text like ? or "address"::text like ?)` as expected.

#### `SearchString` Trait is added. And, `$searchStringColumns` is also added properly.

Also, during installation this warning was thrown, It might be related,
`hoa/ustring suggests installing ext-intl (To get a better Hoa\Ustring::toAscii() and Hoa\Ustring::compareTo().)`

Note: Exact matches works just fine.

**Comment by lorisleiva** [comment]
Hi there, have you defined some columns as `searchable`? (See #17)

**Comment by Niush** [comment]
I missed that completely. It works now, but it is case-sensitive. Is there any way to ignore case or just Lower case search?

**Comment by lorisleiva** [comment]
How would you do this with an SQL query though? 🤔

**Comment by supertassu** [comment]
You can use `ILIKE` instead of `LIKE` to ignore case.

**Comment by Niush** [comment]
**`ILIKE` is only supported by Postgres.**
This commit from spatie query-builder might be useful https://github.com/spatie/laravel-query-builder/pull/70/commits/5863a95c0995bc1089947657aab7e76bda1e10ac

Instead of `$query->where($field, 'LIKE', "%{$solo->content}%");`
Something like this `$query->whereRaw("LOWER({$field}) LIKE ?", ["%{$solo->lower_cased_content}%"]);`

LOWER seems to fail on timestamp column.

**Comment by lorisleiva** [comment]
If it's good enough for Spatie, it's good enough for me. 😅

**Comment by mspiderv** [comment]
Can somebody help me how to "force" usingSearchString to use ILIKE instead of LIKE ? (I use PostgreSQL). Thanks guys.

**Comment by kurucu** [comment]
> **`ILIKE` is only supported by Postgres.**
> This commit from spatie query-builder might be useful [spatie/laravel-query-builder@5863a95](https://github.com/spatie/laravel-query-builder/commit/5863a95c0995bc1089947657aab7e76bda1e10ac)
> 
> Instead of `$query->where($field, 'LIKE', "%{$solo->content}%");`
> Something like this `$query->whereRaw("LOWER({$field}) LIKE ?", ["%{$solo->lower_cased_content}%"]);`
> 
> LOWER seems to fail on timestamp column.

Will this be the fix made? I don't know the package well enough to raise a PR :(

**Comment by lorisleiva** [comment]
Implemented and released in [`v1.1.2`](https://github.com/lorisleiva/laravel-search-string/releases/tag/v1.1.2). 🚀 

Let me know if that all works for you. 🙂
